### DISCUSSION TITLE
How to debug berry crash? Are there source maps?

### DISCUSSION THREAD

**Discussion by stefansundin** [yarnpkg/berry#6061] [General]
Hello,

I'm trying to debug a crash we are experiencing in our CI. I have a suspicion what the problem is, but I want to gather more information before I file a bug report.

Are there source maps available for berry? Right now the stack trace is pretty useless so I'm having trouble pinpointing where the source code is. I think it might even be in a dependency.

Alternatively, is there a way to run an un-minified berry?

Here's my stack trace if anyone wants to help me figure out the problem (version 4.0.2):

```
Error: spawn ETXTBSY
    at ChildProcess.spawn (node:internal/child_process:421:11)
    at Object.spawn (node:child_process:761:9)
    at ZY (/app/.yarn/releases/yarn-4.0.2.cjs:9:51927)
    at RE.implementation (/app/.yarn/releases/yarn-4.0.2.cjs:159:1348)
    at RE.exec (/app/.yarn/releases/yarn-4.0.2.cjs:165:1414)
    at RE.run (/app/.yarn/releases/yarn-4.0.2.cjs:165:1585)
    at ace (/app/.yarn/releases/yarn-4.0.2.cjs:165:7428)
    at async Bot (/app/.yarn/releases/yarn-4.0.2.cjs:167:16)
    at async u (/app/.yarn/releases/yarn-4.0.2.cjs:167:147)
    at async vot (/app/.yarn/releases/yarn-4.0.2.cjs:168:14)
```

We're seeing it happen in CircleCI when running `yarn workspaces focus --production` when building our docker container.

Would love to learn how to debug this. Thanks!

Edit: I think the bug is similar to this one: https://github.com/browserstack/browserstack-runner/pull/217

**Comment by stefansundin** [comment]
Okay, I managed to fix this. We have been pinning the remote docker version in CircleCI and apparently it is super old now.

```yaml
      # DO NOT USE THIS!
      - setup_remote_docker:
          version: 20.10.24
          docker_layer_caching: true
```

So if anyone else is doing this just remove `version` or use `version: default`.

From https://circleci.com/docs/building-docker-images/#docker-version

> To use the current deprecated version, Docker 20, use `20.10.24`

That solved the issue, but perhaps there's an actual bug and this old docker version hits a race condition?

I'm still interested in learning if there are ways to debug berry and how to get a better stack trace.

**Comment by merceyz** [comment]
You're probably running into https://github.com/nodejs/node/issues/48444 so not a bug in Yarn.

You can use [`set version from sources`](https://yarnpkg.com/cli/set/version/from/sources) to get a non-minified version of Yarn:
```sh
yarn set version from sources --no-minify --branch "@yarnpkg/cli/4.0.2"
```

If you want source maps you'll have to clone the repo and build Yarn with

```sh
yarn build:cli --no-minify --source-map
```
then update your project `.yarnrc.yml` to point to the bundle in `./packages/yarnpkg-cli/bundles/yarn.js`.
